yb-voyager/cmd/assessMigrationCommand.go

```diff
@@ -906,6 +914,66 @@ func fetchUnsupportedObjectTypes() ([]UnsupportedFeature, error) {
 	return unsupportedFeatures, nil
 }
 
+func fetchUnsupportedQueryConstructs() ([]utils.UnsupportedQueryConstruct, error) {
+	query := fmt.Sprintf("SELECT DISTINCT query from %s", migassessment.DB_QUERIES_SUMMARY)
+	rows, err := assessmentDB.Query(query)
+	if err != nil {
+		return nil, fmt.Errorf("error querying=%s on assessmentDB: %w", query, err)
+	}
+	defer func() {
+		closeErr := rows.Close()
+		if closeErr != nil {
+			log.Warnf("error closing rows while fetching database queries summary metadata: %v", err)
+		}
+	}()
+
+	var executedQueries []string
+	for rows.Next() {
+		var executedQuery string
+		err := rows.Scan(&executedQuery)
+		if err != nil {
+			return nil, fmt.Errorf("error scanning rows: %w", err)
+		}
+		executedQueries = append(executedQueries, executedQuery)
+	}
+
+	var result []utils.UnsupportedQueryConstruct
+	for i := 0; i < len(executedQueries); i++ {
+		query := executedQueries[i]
+
+		// Check if the query starts with CREATE, INSERT, UPDATE, or DELETE
+		upperQuery := strings.ToUpper(strings.TrimSpace(query))
+		if strings.HasPrefix(upperQuery, "CREATE") || strings.HasPrefix(upperQuery, "INSERT") ||
+			strings.HasPrefix(upperQuery, "UPDATE") || strings.HasPrefix(upperQuery, "DELETE") {
+			continue
+		}
+
+		queryParser := queryparser.New(query)
+		err := queryParser.Parse()
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse query-%s: %w", query, err)
+		}
+
+		// Check for unsupported constructs in the parsed query
+		unsupportedConstructType, err := queryParser.CheckUnsupportedQueryConstruct()
+		if err != nil {
+			log.Warnf("failed while trying to parse the query: %s", err.Error())
+		}
+		if unsupportedConstructType != "" {
+			fmt.Printf("Unsupported query: %s, Type: %s\n", query, unsupportedConstructType)
+			result = append(result, utils.UnsupportedQueryConstruct{
+				ConstructType: unsupportedConstructType,
+				Query:         query,
+			})
+		}
+	}
+	if len(result) != 0 {
+		utils.PrintAndLog("Found YB unsupported queries in source DB, for more details please refer migration assessment report")
```

Comment:
Do we really want to print this as we don't do it for any other things like unsupported features, etc..?

Reply:
Yeah, we can avoid it.
But how about one liner for other also... that way user will need beforehand what all sections in report to look at.